controllers/article.js

```diff
@@ -132,4 +133,44 @@ export default {
     const article = await Article.findOne({ where: { slug } });
     return res.json(article);
   },
+  async bookmarkArticle(req, res) {
+    const { articleId } = req.query;
```

Comment:
To be able to decide what to query article by, you have to check what we return in the article body when we create an article or when we get a list of articles. It is never advisable to try to bookmark or do any other thing with the article when what you need for query (in this case `articleId`) is not naturally available.

Please check the createArticle and getArticle controllers. You'd notice that we always have the slug of an article. Therefore, it is important that `slug` is what you use to find the article in the DB as you are doing in line 137.

If you do this, you'd see that almost all other files require changes too.

It will also mean that what you are doing on lines 78-82 of the bookmark test will not be necessary since slug will be gotten from the responseObject of article creation.

Reply:
Ok. working on that